### PR TITLE
OSS::InformationSection component from 2FA branch

### PR DESCRIPTION
**Pull request by OwenCoogan**
### What does this PR do?

It adds a new component used in the 2FA project

Capture d’écran 2024-12-16 à 15 44 32
Capture d’écran 2024-12-16 à 15 44 13
Capture d’écran 2024-12-16 à 15 36 56
Capture d’écran 2024-12-16 à 12 05 44

Related to: #

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

**Review by Miexil — APPROVED** [review]

File: addon/components/o-s-s/information-section.hbs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<div
+  data-control-name="information-section"
+  class="oss--information-section border-radius-md border-color-gray-200 box-shadow-md"
```
Comment (JulienVannier66) [inline addon/components/o-s-s/information-section.hbs:3]:
```suggestion
  class="oss-information-section border-radius-md border-color-gray-200 box-shadow-md"
```
According to BEM `--` is to modifier, maybe just use `-`

File: addon/components/o-s-s/information-section.hbs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ...attributes
+>
+  <div
+    class="fx-row fx-malign-space-between fx-xalign-center padding-px-18 {{this.plainClass}}"
+    data-control-name="information-section--header"
```
Comment (JulienVannier66) [inline addon/components/o-s-s/information-section.hbs:8]:
```suggestion
    data-control-name="information-section-header"
```
We never use double dash in `data-control-name`, maybe just use single dash

File: addon/components/o-s-s/information-section.hbs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        <OSS::Badge @icon={{@badgeIcon}} class="border-color-primary-200" />
+      {{else if @imageUrl}}
+        <OSS::Badge @image={{@imageUrl}} class="border-color-primary-200" />
+      {{else}}
+        {{yield to="badge"}}
+      {{/if}}
+
+      <div class="fx-col fx-gap-px-3">
+        <span
+          data-control-name="information-section--title"
```
Comment (JulienVannier66) [inline addon/components/o-s-s/information-section.hbs:21]:
```suggestion
          data-control-name="information-section-title"
```

File: addon/components/o-s-s/information-section.hbs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          class="font-color-gray-900 font-size-md font-weight-semibold"
+        >
+          {{@title}}
+        </span>
+
+        {{#if @subtitle}}
+          <span data-control-name="information-section--subtitle" class="font-color-gray-500">{{@subtitle}}</span>
```
Comment (JulienVannier66) [inline addon/components/o-s-s/information-section.hbs:28]:
```suggestion
          <span data-control-name="information-section-subtitle" class="font-color-gray-500">{{@subtitle}}</span>
```

File: addon/components/o-s-s/information-section.hbs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        {{/if}}
+      </div>
+    </div>
+
+    {{yield to="action"}}
+  </div>
+  {{#if (has-block "content")}}
+    <hr class="oss--information-section__separator" />
```
Comment (JulienVannier66) [inline addon/components/o-s-s/information-section.hbs:36]:
```suggestion
    <hr class="oss-information-section__separator" />
```

File: addon/components/o-s-s/information-section.stories.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
+import { hbs } from 'ember-cli-htmlbars';
+
+export default {
+  title: 'Components/OSS::InformationSection',
+  argTypes: {
+    title: {
```
Comment (JulienVannier66) [inline addon/components/o-s-s/information-section.stories.js:6]: For title, according to the TS typing, can you add `type: { required: true }`

File: addon/components/o-s-s/information-section.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  badgeIcon?: keyof typeof IconNames;
+  imageUrl?: string;
+  title: string;
+  subtitle?: string;
+  plain?: boolean;
+}
+
+export default class OSSInformationSection extends Component<InformationSectionArgs> {
+  get plainClass(): string {
+    return this.args.plain === false ? 'background-color-gray-50' : 'background-color-white';
```
Comment (JulienVannier66) [inline addon/components/o-s-s/information-section.ts:14]: For `undefined` value, you want the plain effect? Like the `true` value?

File: addon/components/o-s-s/information-section.hbs [hunk 8]
```diff
@@ -0,0 +1,41 @@
+<div
+  data-control-name="information-section"
+  class="oss--information-section border-radius-md border-color-gray-200 box-shadow-md"
+  ...attributes
+>
+  <div
+    class="fx-row fx-malign-space-between fx-xalign-center padding-px-18 {{this.plainClass}}"
```
Comment (phndiaye) [inline addon/components/o-s-s/information-section.hbs:7]: if we are going to have a getter in it, why not just do a `computedClasses` getter that have all of them + the conditional ones and `Array#join` them?

File: addon/components/o-s-s/information-section.hbs (hunk 1 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.hbs]: Cleaned up

File: addon/components/o-s-s/information-section.hbs (hunk 2 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.hbs]: Cleaned up

File: addon/components/o-s-s/information-section.hbs (hunk 4 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.hbs]: Cleaned up

File: addon/components/o-s-s/information-section.hbs (hunk 5 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.hbs]: Cleaned up

File: addon/components/o-s-s/information-section.ts (hunk 7 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.ts]: Yep, thats it, plain === true would mean the header & content have a white background, plain === false would only have the header in gray. plain === undefined defaults out to all white

File: addon/components/o-s-s/information-section.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.hbs]: Moved all to css

File: addon/components/o-s-s/information-section.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import Component from '@glimmer/component';
+import type { IconNames } from './iconName.enum';
+
+interface InformationSectionArgs {
+  badgeIcon?: keyof typeof IconNames;
```
[the remark this answers is not in the document]
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.ts]: moved to a string type

File: addon/components/o-s-s/information-section.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import Component from '@glimmer/component';
+import type { IconNames } from './iconName.enum';
+
+interface InformationSectionArgs {
```
[the remark this answers is not in the document]
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.ts:3]: Moved mandatory arg to the top

File: addon/components/o-s-s/information-section.hbs
[the remark this answers is not in the document]
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.hbs:1]: left data control name for parent div only

File: addon/components/o-s-s/information-section.hbs (hunk 8 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.hbs]: Didn't see the need to array join but the full string is returned in a headerComputedClasses getter

**Review by phndiaye — APPROVED** [review]

**Pull request merged**